### PR TITLE
Rename concatUnfold to unfoldMany 

### LINKED ISSUE #986: Rename concatUnfold to unfoldMany
This is the unfold counterpart of `foldMany`. This will deprecate the old function.

We can also rename `concatUnfoldInterleave` to `unfoldManyInterleave` and `concatUnfoldRoundrobin` to `unfoldManyRoundRobin`.

### PR DESCRIPTION
**Pull request by rnjtranjan**
Fix #986

### CODE REVIEW

**Review by adithyaov — CHANGES_REQUESTED** [review]
I think you've missed renaming `concatUnfold` to `unfoldMany`. Can you also do that here?

File: Changelog.md [hunk 1]
```diff
@@ -171,7 +171,7 @@
       `Streamly.FileSystem.Handle.read`, `Streamly.Data.Unicode.Stream.decodeUtf8` and
       `splitOnSuffix` with `Streamly.Data.Fold.toList` to split the
        stream to a stream of `String` separated by a newline.
-    * `toHandle` has been deprecated. Please use `intersperse` and `concatUnfold` to
+    * `toHandle` has been deprecated. Please use `intersperse` and `unfoldMany` to
```
Comment (adithyaov) [inline Changelog.md:174]: Don't think we should modify the history of the changelog.

**Review by adithyaov — CHANGES_REQUESTED** [review]
`concatUnfold` to `unfoldMany` is a breaking change.
We should deprecate `concatUnfold` and update the changelog accordingly.
I don't think we should modify the history in the changelog.

File: Changelog.md [hunk 2]
```diff
@@ -247,7 +247,7 @@ concurrently using this.
 
 * Add the following to `Streamly.Prelude`:
     * `unfold`, `fold`, `scan` and `postscan`
-    * `concatUnfold` to concat a stream after unfolding each element
+    * `unfoldMany` to concat a stream after unfolding each element
```
Comment (harendra-kumar) [inline Changelog.md:250]: Revert this change. We need to add a deprecation note in the unreleased section at the top.

File: src/Streamly/Internal/Data/Stream/StreamK/Type.hs [hunk 3]
```diff
@@ -1181,8 +1181,8 @@ instance Monad m => Monad (Stream m) where
 
 {-
 -- Like concatMap but generates stream using an unfold function. Similar to
--- concatUnfold but for StreamK.
-concatUnfoldr :: IsStream t
+-- unfoldMany but for StreamK.
+unfoldManyr :: IsStream t
     => (b -> t m (Maybe (a, b))) -> t m b -> t m a
-concatUnfoldr = undefined
+unfoldManyr = undefined
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Stream/StreamK/Type.hs:1187]: We can revert this renaming, this is more of a concatUnfoldr.

File: src/Streamly/Internal/Data/Stream/IsStream/Expand.hs [hunk 4]
```diff
@@ -460,29 +460,29 @@ mergeAsyncByM f m1 m2 =
 -- therefore provide many times better performance.
 --
 -- @since 0.7.0
-{-# INLINE concatUnfold #-}
-concatUnfold ::(IsStream t, Monad m) => Unfold m a b -> t m a -> t m b
-concatUnfold u m = fromStreamD $ D.concatUnfold u (toStreamD m)
+{-# INLINE unfoldMany #-}
+unfoldMany ::(IsStream t, Monad m) => Unfold m a b -> t m a -> t m b
+unfoldMany u m = fromStreamD $ D.unfoldMany u (toStreamD m)
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Stream/IsStream/Expand.hs:465]: We need to retain the old function as it is, just add a `DEPRECATED` pragma. Introduce a new `unfoldMany` function and move the doc from `concatUnfold` to the new function, and change the since notation to 0.8.0. In `concatUnfold` just say "Same as unfoldMany".

File: src/Streamly/Prelude.hs [hunk 5]
```diff
@@ -595,7 +595,7 @@ module Streamly.Prelude
     , concatMapWith
     , concatMap
     , concatMapM
-    , concatUnfold
+    , unfoldMany
```
Comment (harendra-kumar) [inline src/Streamly/Prelude.hs:598]: We need to keep concatUnfold as well, it is only deprecated.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
